numpy/lib/utils.py

```diff
@@ -587,7 +587,8 @@ def info(object=None, maxwidth=76, output=sys.stdout, toplevel='numpy'):
             print(inspect.getdoc(object), file=output)
 
         methods = pydoc.allmethods(object)
-        if methods != []:
+
+        if any(meth[0] != '_' for meth in methods):
             print("\n\nMethods:\n", file=output)
             for meth in methods:
                 if meth[0] == '_':
```

Comment:
Perhaps better as:
```
public_methods = [meth for meth in methods if meth[0] != '_']
if public_methods:
    ...
```

Reply:
Thanks, that definitely improves readability. If we have the list `public_methods` available, might as well change this
```python
for meth in methods:
    if meth[0] == '_':
        continue
```
to this
```python
for meth in public_methods:
```
since the `meth[0] == '_'` check is redundant.
Do you agree?

Reply:
That's what I was thinking :)

Reply:
Great! I just updated it.